server/app/app.py

```diff
@@ -1,26 +1,190 @@
 import os
 import datetime
 
-from flask import Flask
+from flask import Flask, redirect, current_app, make_response, render_template
+from flask import Blueprint, request, send_from_directory
 from flask_caching import Cache
 from flask_compress import Compress
 from flask_cors import CORS
+from flask_restful import Api, Resource
 
-from server.app.rest_api.rest import get_api_resources
-from server.app.util.utils import Float32JSONEncoder
-from server.app.web import webapp
+from http import HTTPStatus
+
+import server.common.rest as common_rest
+from server.common.utils import path_join, Float32JSONEncoder
+from server.data_common.matrix_loader import MatrixDataLoader, MatrixDataType
+
+from functools import wraps
+
+webbp = Blueprint("webapp", "server.common.web", template_folder="templates")
+
+
+@webbp.route("/")
+def index():
+    if current_app.app_config.datapath is None:
+        return dataroot_index()
+
+    dataset_title = current_app.app_config.title
```

Comment:
shouldn't this call dataset_index(), rather than duplicating the template setup and render?

Reply:
fixed